ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -547,6 +564,48 @@ filterNotAlreadyInFlightWithOtherPeers FetchModeBulkSync chains =
             _other                  -> peerFetchBlocksInFlight inflight
         | (_, status, inflight, _) <- chains ]
 
+    -- The highest slot number that is or has been in flight for any peer.
+    maxSlotNoInFlightWithOtherPeers = maximum $ Nothing :
+      [ peerFetchMaxSlotNo inflight | (_, _, inflight, _) <- chains ]
+
+-- | Filter a fragment, but stop the filtering (and add the remaining fragment
```

Comment:
Let's say something like: this is intended to an optimized but equivalent form of `filter`. For this to be the case, the following precondition needs to be satisfied: (...).

Reply:
And it might be nice to have a test that captures that particular use case (where `filterWithStop` is in fact equivalent to `fitler`, with a non-trivial predicate).